### PR TITLE
Unpin and upgrade dependencies

### LINKED ISSUE #262: CVE-2023-45133
There's a @babel/traverse issue [being reported](https://security.snyk.io/vuln/SNYK-JS-BABELTRAVERSE-5962462).
Your package json forces "old" version of it, the fix is available.

### PR DESCRIPTION
**Pull request by harryzcy**
Upgrade `@babel-traverse` to non-vulnerable version 7.23.2. CVE-2023-45133
Unpin dependencies to permit future upgrades, without changing prettier-plugin-sort-imports.

Fix #262

### CODE REVIEW

**Comment by richardjelinek-fastest** [comment]
🙏🥺

File: src/utils/__tests__/get-code-from-ast.spec.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ import a from 'a';
     expect(format(formatted, { parser: 'babel' })).toEqual(
         `// first comment
 // second comment
+
```
Comment (martibenn) [inline src/utils/__tests__/get-code-from-ast.spec.ts:33]: I don't see the point adding new lines ? Is this coming from an IDE formater ?

Comment (marklai1998) [reply src/utils/__tests__/get-code-from-ast.spec.ts:33]: no, for some reason the snapshot doesn't match with those new line

Comment (harryzcy) [reply src/utils/__tests__/get-code-from-ast.spec.ts:33]: They are generated after `@babel/generator` >=7.19.

**Review by marklai1998 — APPROVED** [review]

**Review by martibenn — APPROVED** [review]

**Comment by harryzcy** [comment]
@ayusharma Could you help with reviewing this?

**Comment by marklai1998** [comment]
@byara

**Review by byara — COMMENTED** [review]
Thank you for your contribution. Other than the new lines added to the snapshots, this looks good to me. 
Can we figure out a way not to add those new lines? Otherwise, this is a breaking change

File: src/utils/__tests__/get-code-from-ast.spec.ts (hunk 1 above)
Comment (hamidrezahanafi) [reply src/utils/__tests__/get-code-from-ast.spec.ts:33]: @byara It seems that they are auto-generated by `babel/generator`

**Comment by harryzcy** [comment]
> Thank you for your contribution. Other than the new lines added to the snapshots, this looks good to me. Can we figure out a way not to add those new lines? Otherwise, this is a breaking change

I couldn't find any `@babel/generator` that controls it.

File: src/utils/__tests__/get-code-from-ast.spec.ts (hunk 1 above)
Comment (Eldemarkki) [reply src/utils/__tests__/get-code-from-ast.spec.ts:33]: With 7.19.3 the tests pass, but 7.19.4 they fail. Probably due to this PR: https://github.com/babel/babel/pull/14979

Comment (marklai1998) [reply src/utils/__tests__/get-code-from-ast.spec.ts:33]: @Eldemarkki The PR tagged as 8.0.0-alpha.2 but the changelog saids 7.19.4.
Did they just messed up the release?

Comment (marklai1998) [reply src/utils/__tests__/get-code-from-ast.spec.ts:33]: seems its an unvoidable breaking change

**Comment by harryzcy** [comment]
@byara #266 only updates `@babel/traverse` and leaves other dependencies untouched. 

That could be the non-breaking one. And this PR could lead to a breaking change. What do you think?

**Comment by byara** [comment]
The change for babel traverse is release in [v4.2.1](https://github.com/trivago/prettier-plugin-sort-imports/releases/tag/v4.2.1)

**Comment by marklai1998** [comment]
@byara any updates?

**Comment by byara** [comment]
This should be addressed in the [v5](https://github.com/trivago/prettier-plugin-sort-imports/releases/tag/v5.0.0). Please feel free to reopen in there is an issue.

**Comment by StavNoyAkur8** [comment]
> This should be addressed in the [v5](https://github.com/trivago/prettier-plugin-sort-imports/releases/tag/v5.0.0). Please feel free to reopen in there is an issue.

v5 looks to have updated the dependencies **but kept them pinned**.
https://github.com/trivago/prettier-plugin-sort-imports/blob/35d7b0b0ec7f277bfc13d4cd4be92d4f0867ba8f/package.json#L36-L41
That means next time there's a vulnerability found, we  will again need to wait for an update here, or override the dependencies.

I can't reopen, since it's not my PR.

**Comment by byara** [comment]
FYI @StavNoyAkur8 #322

**Pull request closed without merging**